src/com/xilinx/rapidwright/edif/EDIFNet.java

```diff
@@ -71,13 +71,27 @@ protected EDIFNet(){
 	 */
 	public void addPortInst(EDIFPortInst portInst){
 		if(portInsts == null) portInsts = new EDIFPortInstList();
-		if(parentCell != null && portInst.getCellInst() == null) {
+		boolean isParentCellNonNull = parentCell != null;
+		EDIFCellInst inst = portInst.getCellInst();
+		if(isParentCellNonNull && inst == null) {
 			parentCell.addInternalPortMapEntry(portInst.getName(), this);
 		}
 		portInst.setParentNet(this);
+		if(isParentCellNonNull) {
+		    trackChanges(EDIFChangeType.PORT_INST_ADD, inst, portInst.getName());
```

Comment:
```suggestion
		    // This does not explicitly track the port instance index, in most cases the name should be sufficient.
		    trackChanges(EDIFChangeType.PORT_INST_ADD, inst, portInst.getName());
```

Reply:
Actually, this isn't true for the ADD case. If we've added a port inst, then we can query its index (which is what I have been doing). Unless of course, that port inst is later REMOVE-d.

Reply:
If the only record used to track a design change is an `EDIFChangeNet` object, it is not tracked explicitly.

Reply:
`EDIFChangeNet` tracks net and instance, and then from the inst I can find the port inst via its name, and thus find the index?